### PR TITLE
Use Logger for better debugging

### PR DESCRIPTION
**Pull request by liudangyi**
Add `--debug` flag to `~/.ipython/kernels/ruby/kernel.json` to turn on debug info.

### CODE REVIEW

File: lib/iruby/kernel.rb [hunk 1]
```diff
@@ -33,8 +36,8 @@ def initialize(config_file)
           hb_socket = c.socket(ZMQ::REP)
           hb_socket.bind(connection % @config['hb_port'])
           ZMQ::Device.new(hb_socket, hb_socket)
-        rescue Exception => ex
-          STDERR.puts "Kernel heartbeat died: #{ex.message}\n"#{ex.backtrace.join("\n")}"
```
Comment (minad) [inline lib/iruby/kernel.rb]: No, we should catch exception here. Otherwise the thread will just die without a word.

**Comment by minad** [comment]
Please don't mix different things in a single pr. Makes it harder to review.

The policy not to catch exceptions holds only for library code etc. It doesn't hold for the code at the very top of the call stack, e.g. the loop of a repl - this should definitely catch exceptions.

File: lib/iruby/kernel.rb (hunk 1 above)
Comment (liudangyi) [inline lib/iruby/kernel.rb]: I think we have already catch exception [in command.rb](https://github.com/SciRuby/iruby/pull/12/files#diff-9d5692ae3ea9c349b6d32a29d63fb1eeL62) so maybe it's unnecessary to catch them here.

**Comment by liudangyi** [comment]
> Please don't mix different things in a single pr. Makes it harder to review.

Sorry for that.

File: lib/iruby/kernel.rb (hunk 1 above)
Comment (minad) [inline lib/iruby/kernel.rb]: No this is a thread. Please try it yourself. Create a thread and throw an exception and see what happens.

**Comment by minad** [comment]
I close this one, please reopen a new one.

**Pull request closed without merging**